From 52f39ce7d3969fdd699d7f9293fcd1cd33504b4d Mon Sep 17 00:00:00 2001
From: JRoy <joshroy126@gmail.com>
Date: Sat, 2 May 2020 15:34:17 -0400
Subject: [PATCH] Add AccountOwnerCheckEvent

Allows for custom sign owner handling

diff --git a/src/main/java/com/Acrobot/ChestShop/Events/AccountOwnerCheckEvent.java b/src/main/java/com/Acrobot/ChestShop/Events/AccountOwnerCheckEvent.java
new file mode 100644
index 0000000..d4f632c
--- /dev/null
+++ b/src/main/java/com/Acrobot/ChestShop/Events/AccountOwnerCheckEvent.java
@@ -0,0 +1,46 @@
+package com.Acrobot.ChestShop.Events;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+public class AccountOwnerCheckEvent extends Event implements Cancellable {
+  private static final HandlerList handlers = new HandlerList();
+
+  private String name;
+  private Player player;
+  private boolean b = false;
+
+  public AccountOwnerCheckEvent(Player player, String name) {
+    this.player = player;
+    this.name = name;
+  }
+
+  public Player getPlayer() {
+    return player;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public HandlerList getHandlers() {
+    return handlers;
+  }
+
+  public static HandlerList getHandlerList() {
+    return handlers;
+  }
+
+  //Cancelled event means player does not have access.
+  @Override
+  public boolean isCancelled() {
+    return b;
+  }
+
+  @Override
+  public void setCancelled(boolean b) {
+    this.b = b;
+  }
+}
\ No newline at end of file
diff --git a/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java b/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java
index 1f3aff4..0a1b668 100644
--- a/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java
+++ b/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java
@@ -375,6 +375,12 @@ public class NameManager implements Listener {
             return true;
         }
 
+        //ChestShopPlus Start
+        com.Acrobot.ChestShop.Events.AccountOwnerCheckEvent accountOwnerCheckEvent = new com.Acrobot.ChestShop.Events.AccountOwnerCheckEvent(player, name);
+        ChestShop.callEvent(accountOwnerCheckEvent);
+        if (accountOwnerCheckEvent.isCancelled()) return true;
+        if (name.startsWith("#")) return false;
+        //ChestShopPlus End
         AccountQueryEvent queryEvent = new AccountQueryEvent(name);
         queryEvent.searchOfflinePlayers(false);
         ChestShop.callEvent(queryEvent);
-- 
2.20.1.windows.1

