From 1d594fb61752f91866595d82b7813f6f97d2e65b Mon Sep 17 00:00:00 2001
From: JRoy <joshroy126@gmail.com>
Date: Sat, 2 May 2020 15:35:35 -0400
Subject: [PATCH] Add PreCurrencyAddEvent


diff --git a/src/main/java/com/Acrobot/ChestShop/Events/PreCurrencyAddEvent.java b/src/main/java/com/Acrobot/ChestShop/Events/PreCurrencyAddEvent.java
new file mode 100644
index 0000000..d68805a
--- /dev/null
+++ b/src/main/java/com/Acrobot/ChestShop/Events/PreCurrencyAddEvent.java
@@ -0,0 +1,47 @@
+package com.Acrobot.ChestShop.Events;
+
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+public class PreCurrencyAddEvent extends Event implements Cancellable {
+  private static final HandlerList handlers = new HandlerList();
+
+  private UUID target;
+  private BigDecimal amountSent;
+  private boolean b = false;
+
+  public PreCurrencyAddEvent(UUID target, BigDecimal amountSent) {
+    this.target = target;
+    this.amountSent = amountSent;
+  }
+
+  public UUID getTarget() {
+    return target;
+  }
+
+  public BigDecimal getAmountSent() {
+    return amountSent;
+  }
+
+  public HandlerList getHandlers() {
+    return handlers;
+  }
+
+  public static HandlerList getHandlerList() {
+    return handlers;
+  }
+
+  @Override
+  public boolean isCancelled() {
+    return b;
+  }
+
+  @Override
+  public void setCancelled(boolean b) {
+    this.b = b;
+  }
+}
\ No newline at end of file
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java b/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java
index e7e5954..670dacd 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java
@@ -203,6 +203,15 @@ public class VaultListener extends EconomyAdapter {
             return;
         }
 
+        //ChestShopPlus Start
+        com.Acrobot.ChestShop.Events.PreCurrencyAddEvent preCurrencyAddEvent = new com.Acrobot.ChestShop.Events.PreCurrencyAddEvent(event.getTarget(), event.getAmount());
+        ChestShop.callEvent(preCurrencyAddEvent);
+        if(preCurrencyAddEvent.isCancelled()) {
+            event.setAdded(true);
+            return;
+        }
+        //ChestShopPlus End
+
         World world = event.getWorld();
         //String lastSeen = NameManager.getLastSeenName(event.getTarget());
         OfflinePlayer lastSeen = Bukkit.getOfflinePlayer(event.getTarget());
-- 
2.20.1.windows.1

