From bca64dc916ae6bd1cd4cb7d23096e68e8c05941c Mon Sep 17 00:00:00 2001
From: JRoy <joshroy126@gmail.com>
Date: Sat, 2 May 2020 15:36:22 -0400
Subject: [PATCH] Add sign stock indication color


diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/Block/Break/SignBreak.java b/src/main/java/com/Acrobot/ChestShop/Listeners/Block/Break/SignBreak.java
index 23c5398..615d39e 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/Block/Break/SignBreak.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/Block/Break/SignBreak.java
@@ -142,7 +142,7 @@ public class SignBreak implements Listener {
                 continue;
             }
 
-            if (Properties.TURN_OFF_SIGN_PROTECTION || canDestroyShop(breaker, sign.getLine(NAME_LINE))) {
+            if (Properties.TURN_OFF_SIGN_PROTECTION || canDestroyShop(breaker, org.bukkit.ChatColor.stripColor(sign.getLine(NAME_LINE)))) { //ChestShopPlus - Sign stock color
                 brokenBlocks.add(sign);
             } else {
                 canBeBroken = false;
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/Modules/DiscountModule.java b/src/main/java/com/Acrobot/ChestShop/Listeners/Modules/DiscountModule.java
index 2b1cbd9..036d700 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/Modules/DiscountModule.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/Modules/DiscountModule.java
@@ -62,7 +62,7 @@ public class DiscountModule implements Listener {
 
         Player client = event.getClient();
 
-        if (!PriceUtil.hasBuyPrice(event.getSign().getLine(PRICE_LINE))) {
+        if (!PriceUtil.hasBuyPrice(org.bukkit.ChatColor.stripColor(event.getSign().getLine(PRICE_LINE)))) { //ChestShopPlus - Sign stock color
             return;
         }
 
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java b/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java
index dc18afc..7fdb3a3 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java
@@ -153,10 +153,10 @@ public class PlayerInteract implements Listener {
     }
 
     private static PreTransactionEvent preparePreTransactionEvent(Sign sign, Player player, Action action) {
-        String name = sign.getLine(NAME_LINE);
-        String quantity = sign.getLine(QUANTITY_LINE);
-        String prices = sign.getLine(PRICE_LINE);
-        String material = sign.getLine(ITEM_LINE);
+        String name = org.bukkit.ChatColor.stripColor(sign.getLine(NAME_LINE)); //ChestShopPlus - Sign stock color
+        String quantity = org.bukkit.ChatColor.stripColor(sign.getLine(QUANTITY_LINE)); //ChestShopPlus - Sign stock color
+        String prices = org.bukkit.ChatColor.stripColor(sign.getLine(PRICE_LINE)); //ChestShopPlus - Sign stock color
+        String material = org.bukkit.ChatColor.stripColor(sign.getLine(ITEM_LINE)); //ChestShopPlus - Sign stock color
 
         AccountQueryEvent accountQueryEvent = new AccountQueryEvent(name);
         Bukkit.getPluginManager().callEvent(accountQueryEvent);
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/PostShopCreation/SignSticker.java b/src/main/java/com/Acrobot/ChestShop/Listeners/PostShopCreation/SignSticker.java
index 1b28ff0..06e455e 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/PostShopCreation/SignSticker.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/PostShopCreation/SignSticker.java
@@ -26,7 +26,7 @@ public class SignSticker implements Listener {
             return;
         }
 
-        if (ChestShopSign.isAdminShop(event.getSignLine(NAME_LINE))) {
+        if (ChestShopSign.isAdminShop(org.bukkit.ChatColor.stripColor(event.getSignLine(NAME_LINE)))) { //ChestShopPlus - Sign stock color
             return;
         }
 
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/PostTransaction/EconomicModule.java b/src/main/java/com/Acrobot/ChestShop/Listeners/PostTransaction/EconomicModule.java
index c4b8eb4..1819c97 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/PostTransaction/EconomicModule.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/PostTransaction/EconomicModule.java
@@ -24,6 +24,20 @@ public class EconomicModule implements Listener {
         ChestShop.callEvent(currencyTransferEvent);
         if (!currencyTransferEvent.wasHandled()) {
             event.setCancelled(true);
+            return; //ChestShopPlus - Sign stock color
         }
+        //ChestShopPlus Start - Sign stock color
+        if (event.getTransactionType() == BUY) {
+            int count = 0;
+            for (org.bukkit.inventory.ItemStack stack : event.getOwnerInventory().getContents()) {
+                if (stack != null && stack.getType() == event.getStock()[0].getType()) {
+                    count += stack.getAmount();
+                }
+            }
+            com.Acrobot.ChestShop.Utils.uBlock.updateSignText(event.getSign(), count > 1);
+        } else {
+            com.Acrobot.ChestShop.Utils.uBlock.updateSignText(event.getSign(), true);
+        }
+        //ChestShopPlus End - Sign stock color
     }
 }
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/PreTransaction/PartialTransactionModule.java b/src/main/java/com/Acrobot/ChestShop/Listeners/PreTransaction/PartialTransactionModule.java
index 40502b8..b9b752a 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/PreTransaction/PartialTransactionModule.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/PreTransaction/PartialTransactionModule.java
@@ -69,12 +69,14 @@ public class PartialTransactionModule implements Listener {
 
             if (possessedItemCount <= 0) {
                 event.setCancelled(NOT_ENOUGH_STOCK_IN_CHEST);
+                com.Acrobot.ChestShop.Utils.uBlock.updateSignText(event.getSign(), false); //ChestShopPlus - Sign stock color
                 return;
             }
 
             event.setExactPrice(pricePerItem.multiply(BigDecimal.valueOf(possessedItemCount)).setScale(Properties.PRICE_PRECISION, BigDecimal.ROUND_HALF_UP));
             event.setStock(itemsHad);
         }
+        com.Acrobot.ChestShop.Utils.uBlock.updateSignText(event.getSign(), true); //ChestShopPlus - Sign stock color
 
         if (!InventoryUtil.fits(event.getStock(), event.getClientInventory())) {
             ItemStack[] itemsFit = getItemsThatFit(event.getStock(), event.getClientInventory());
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRefundListener.java b/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRefundListener.java
index 78e5761..3593903 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRefundListener.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRefundListener.java
@@ -41,7 +41,7 @@ public class ShopRefundListener implements Listener {
             return;
         }
 
-        AccountQueryEvent accountQueryEvent = new AccountQueryEvent(event.getSign().getLine(NAME_LINE));
+        AccountQueryEvent accountQueryEvent = new AccountQueryEvent(org.bukkit.ChatColor.stripColor(event.getSign().getLine(NAME_LINE))); //ChestShopPlus - Sign stock color
         Bukkit.getPluginManager().callEvent(accountQueryEvent);
         Account account = accountQueryEvent.getAccount();
         if (account == null) {
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRemovalLogger.java b/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRemovalLogger.java
index 1174eb3..4c0ff44 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRemovalLogger.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/ShopRemoval/ShopRemovalLogger.java
@@ -25,11 +25,11 @@ public class ShopRemovalLogger implements Listener {
 
         ChestShop.getBukkitServer().getScheduler().runTaskAsynchronously(ChestShop.getPlugin(), new Runnable() {
             @Override public void run() {
-                String shopOwner = event.getSign().getLine(NAME_LINE);
+                String shopOwner = org.bukkit.ChatColor.stripColor(event.getSign().getLine(NAME_LINE)); //ChestShopPlus - Sign stock color
                 String typeOfShop = ChestShopSign.isAdminShop(shopOwner) ? "An Admin Shop" : "A shop belonging to " + shopOwner;
 
-                String item = event.getSign().getLine(QUANTITY_LINE) + ' ' + event.getSign().getLine(ITEM_LINE);
-                String prices = event.getSign().getLine(PRICE_LINE);
+                String item = org.bukkit.ChatColor.stripColor(event.getSign().getLine(QUANTITY_LINE)) + ' ' + org.bukkit.ChatColor.stripColor(event.getSign().getLine(ITEM_LINE)); //ChestShopPlus - Sign stock color
+                String prices = org.bukkit.ChatColor.stripColor(event.getSign().getLine(PRICE_LINE)); //ChestShopPlus - Sign stock color
                 String location = LocationUtil.locationToString(event.getSign().getLocation());
 
                 String message = String.format(REMOVAL_MESSAGE,
diff --git a/src/main/java/com/Acrobot/ChestShop/Security.java b/src/main/java/com/Acrobot/ChestShop/Security.java
index 9c2b611..400b7de 100644
--- a/src/main/java/com/Acrobot/ChestShop/Security.java
+++ b/src/main/java/com/Acrobot/ChestShop/Security.java
@@ -89,7 +89,7 @@ public class Security {
                 continue;
             }
 
-            AccountQueryEvent accountQueryEvent = new AccountQueryEvent(sign.getLine(ChestShopSign.NAME_LINE));
+            AccountQueryEvent accountQueryEvent = new AccountQueryEvent(org.bukkit.ChatColor.stripColor(sign.getLine(ChestShopSign.NAME_LINE))); //ChestShopPlus - Sign stock color
             Bukkit.getPluginManager().callEvent(accountQueryEvent);
             Account account = accountQueryEvent.getAccount();
             if (account != null && !account.getUuid().equals(player.getUniqueId())) {
diff --git a/src/main/java/com/Acrobot/ChestShop/Signs/ChestShopSign.java b/src/main/java/com/Acrobot/ChestShop/Signs/ChestShopSign.java
index 486162f..cdbadd6 100644
--- a/src/main/java/com/Acrobot/ChestShop/Signs/ChestShopSign.java
+++ b/src/main/java/com/Acrobot/ChestShop/Signs/ChestShopSign.java
@@ -120,7 +120,7 @@ public class ChestShopSign {
         if (player == null) return false;
         if (sign == null) return true;
 
-        String name = sign.getLine(NAME_LINE);
+        String name = org.bukkit.ChatColor.stripColor(sign.getLine(NAME_LINE)); //ChestShopPlus - Sign stock color
         if (name == null || name.isEmpty()) return true;
 
         return NameManager.canUseName(player, base, name);
@@ -129,7 +129,7 @@ public class ChestShopSign {
     public static boolean isOwner(Player player, Sign sign) {
         if (player == null || sign == null) return false;
 
-        String name = sign.getLine(NAME_LINE);
+        String name = org.bukkit.ChatColor.stripColor(sign.getLine(NAME_LINE)); //ChestShopPlus - Sign stock color
         if (name == null || name.isEmpty()) return false;
 
         AccountQueryEvent accountQueryEvent = new AccountQueryEvent(name);
diff --git a/src/main/java/com/Acrobot/ChestShop/Utils/uBlock.java b/src/main/java/com/Acrobot/ChestShop/Utils/uBlock.java
index 4b39efe..26911c3 100644
--- a/src/main/java/com/Acrobot/ChestShop/Utils/uBlock.java
+++ b/src/main/java/com/Acrobot/ChestShop/Utils/uBlock.java
@@ -221,4 +221,16 @@ public class uBlock {
     public static boolean couldBeShopContainer(InventoryHolder holder) {
         return holder instanceof Container && couldBeShopContainer(((Container) holder).getBlock());
     }
+
+    //ChestShopPlus Start - Sign stock color
+    public static void updateSignText(Sign sign, boolean inStock) {
+        if (org.bukkit.ChatColor.stripColor(sign.getLine(0)).equalsIgnoreCase("Admin Shop")) {
+            return;
+        }
+        for (int i = 0; i < 4; i++) {
+            sign.setLine(i, (inStock ? "" : org.bukkit.ChatColor.RED) + org.bukkit.ChatColor.stripColor(sign.getLine(i)));
+        }
+        sign.update();
+    }
+    //ChestShopPlus End - Sign stock color
 }
-- 
2.20.1.windows.1

