From 98903394039008323f4b825b4a698d8ce83a9485 Mon Sep 17 00:00:00 2001
From: JRoy <joshroy126@gmail.com>
Date: Sat, 2 May 2020 15:36:36 -0400
Subject: [PATCH] Add PreAccountCheckEvent


diff --git a/src/main/java/com/Acrobot/ChestShop/Events/Economy/AccountCheckEvent.java b/src/main/java/com/Acrobot/ChestShop/Events/Economy/AccountCheckEvent.java
index 2307ce5..5d11dd5 100644
--- a/src/main/java/com/Acrobot/ChestShop/Events/Economy/AccountCheckEvent.java
+++ b/src/main/java/com/Acrobot/ChestShop/Events/Economy/AccountCheckEvent.java
@@ -12,6 +12,7 @@ import java.util.UUID;
  * @author Acrobot
  */
 public class AccountCheckEvent extends Event {
+    private com.Acrobot.ChestShop.Database.Account accountObj; //ChestShopPlus - PreAccountCheckEvent
     private static final HandlerList handlers = new HandlerList();
 
     boolean outcome = false;
@@ -65,4 +66,14 @@ public class AccountCheckEvent extends Event {
     public static HandlerList getHandlerList() {
         return handlers;
     }
+
+    //ChestShopPlus Start
+    public void setAccountObj(com.Acrobot.ChestShop.Database.Account accountObj) {
+        this.accountObj = accountObj;
+    }
+
+    public com.Acrobot.ChestShop.Database.Account getAccountObj() {
+        return accountObj;
+    }
+    //ChestShopPlus End
 }
diff --git a/src/main/java/com/Acrobot/ChestShop/Events/PreAccountCheckEvent.java b/src/main/java/com/Acrobot/ChestShop/Events/PreAccountCheckEvent.java
new file mode 100644
index 0000000..6b5c128
--- /dev/null
+++ b/src/main/java/com/Acrobot/ChestShop/Events/PreAccountCheckEvent.java
@@ -0,0 +1,43 @@
+package com.Acrobot.ChestShop.Events;
+
+import com.Acrobot.ChestShop.Database.Account;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+public class PreAccountCheckEvent extends Event implements Cancellable {
+  private static final HandlerList handlers = new HandlerList();
+
+  private Account account;
+
+  private boolean b = false;
+
+  public PreAccountCheckEvent(Account account) {
+    this.account = account;
+  }
+
+  public Account getAccount() {
+    return account;
+  }
+
+  public HandlerList getHandlers() {
+    return handlers;
+  }
+
+  public static HandlerList getHandlerList() {
+    return handlers;
+  }
+
+  @Override
+  public boolean isCancelled() {
+    return b;
+  }
+
+  @Override
+  public void setCancelled(boolean b) {
+    this.b = b;
+  }
+}
\ No newline at end of file
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java b/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java
index 532e995..ea1db27 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/Economy/Plugins/VaultListener.java
@@ -170,6 +170,15 @@ public class VaultListener extends EconomyAdapter {
             return;
         }
 
+        //ChestShopPlus Start
+        com.Acrobot.ChestShop.Events.PreAccountCheckEvent preAccountCheckEvent = new com.Acrobot.ChestShop.Events.PreAccountCheckEvent(event.getAccountObj());
+        ChestShop.callEvent(preAccountCheckEvent);
+        if (preAccountCheckEvent.isCancelled()) {
+            event.hasAccount(true);
+            return;
+        }
+        //ChestShopPlus End
+
         World world = event.getWorld();
         //String lastSeen = NameManager.getLastSeenName(event.getAccount());
         OfflinePlayer lastSeen = Bukkit.getOfflinePlayer(event.getAccount());
diff --git a/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java b/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java
index 42e0f29..14399f4 100644
--- a/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java
+++ b/src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java
@@ -166,6 +166,7 @@ public class PlayerInteract implements Listener {
         // check if player exists in economy
         if (!adminShop) {
             AccountCheckEvent event = new AccountCheckEvent(account.getUuid(), player.getWorld());
+            event.setAccountObj(account); //ChestShopPlus - PreAccountCheckEvent
             Bukkit.getPluginManager().callEvent(event);
             if(!event.hasAccount()) {
                 player.sendMessage(Messages.prefix(Messages.NO_ECONOMY_ACCOUNT));
-- 
2.20.1.windows.1

