From c1dde264c36bb8c1c40729bf8cbf75ae946eee4f Mon Sep 17 00:00:00 2001
From: JRoy <joshroy126@gmail.com>
Date: Sat, 2 May 2020 15:37:54 -0400
Subject: [PATCH] Add NameManager#createAccount method

Allows for the creation of accounts in other plugins

diff --git a/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java b/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java
index 17459d9..6974046 100644
--- a/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java
+++ b/src/main/java/com/Acrobot/ChestShop/UUIDs/NameManager.java
@@ -287,6 +287,18 @@ public class NameManager implements Listener {
         return null;
     }
 
+    //ChestShopPlus Start
+    public static Account createAccount(Account account) {
+        try {
+            accounts.createOrUpdate(account);
+        } catch (SQLException e) {
+            ChestShop.getBukkitLogger().log(Level.WARNING, "Error while updating account " + account + ":", e);
+            return null;
+        }
+        return account;
+    }
+    //ChestShopPlus End
+
     /**
      * Get the short username from a full username
      *
-- 
2.20.1.windows.1

